### PR TITLE
`Overlay`: make aria compliant, `QTip`: remove component

### PR DESCRIPTION
**Pull request by ArnaudWeyts**
### Description

This is the first part of making our `Popover` and `Dialog` components aria compliant and keyboard navigable.

### Fixed

- `Overlay`: Pressing ESC now closes the relevant overlay, instead of all overlays
- `DatePicker`: `formatDate` util function signature

### Changed

- [BREAKING] `Overlay`: The Overlay component should contain children instead of being positioned adjacent to children
- `Overlay`: lowered the color opacity of the overlay ([@ArnaudWeyts](https://github.com/ArnaudWeyts) in [#1729](https://github.com/teamleadercrm/ui/pull/1729))

### Removed

- [BREAKING] `QTip`: This component has been removed

### Manual check

- [x] Dialogs are still displayed with a background and can be dismissed properly
- [x] Popovers are still displayed correctly and can be dismissed properly
  - [x] Also check the DatePicker components
  - [x] Also check the Passport components
  - [x] Also check the SplitButton component
  - [x] Also check the Wysywig editor link component

### CODE REVIEW

**Review by lorgan3 — APPROVED** [review]

**Review by EOSullivanBerlin — APPROVED** [review]

**Review by adognameddog — APPROVED** [review]

File: src/components/overlay/theme.css [hunk 1]
```diff
@@ -13,6 +13,6 @@
 
 .is-entered {
   &.dark {
-    background-color: color(var(--color-teal-dark) a(85%));
+    background-color: color(var(--color-teal-dark) a(50%));
```
Comment (lowiebenoot) [inline src/components/overlay/theme.css:16]: Is this decided by a designer? I don't see this change in figma.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,15 +12,21 @@
 
 ### Changed
 
+- [BREAKING] `Overlay`: The Overlay component should contain children instead of being positioned adjacent to children ([@ArnaudWeyts](https://github.com/ArnaudWeyts) in [#1729](https://github.com/teamleadercrm/ui/pull/1729))
```
Comment (lowiebenoot) [inline CHANGELOG.md:15]: I guess we should also mention the removal of these props
```
onOverlayMouseDown
onOverlayMouseMove
onOverlayMouseUp,
```

**Review by lowiebenoot — APPROVED** [review]

**Comment by lowiebenoot** [comment]
Checked ✅

File: src/components/overlay/theme.css (hunk 1 above)
Comment (ArnaudWeyts) [reply src/components/overlay/theme.css:16]: Yup, @driesd told me to implement this, and he was told by our resident designer Stephen 🙂

File: CHANGELOG.md (hunk 2 above)
Comment (ArnaudWeyts) [reply CHANGELOG.md:15]: Good point!

**Review by driesd — APPROVED** [review]

**Review by lowiebenoot — APPROVED** [review]

**Pull request merged**